components/data-provider/src/structs/mod.rs

```diff
@@ -0,0 +1,19 @@
+pub mod decimal;
+pub mod plurals;
+
+#[cfg(feature = "invariant")]
+use crate::prelude::*;
+
+/// Gets a locale-invariant default struct given a data key in this module's category.
+/// For example, if the data key is `plurals/cardinal@1`, a Response with an object of type
+/// PluralRuleStringsV1 will be returned.
+#[cfg(feature = "invariant")]
+pub(crate) fn get_invariant(data_key: &DataKey) -> Option<DataResponse<'static>> {
+    if let Some(response) = decimal::get_invariant(data_key) {
+        Some(response)
+    } else if let Some(response) = plurals::get_invariant(data_key) {
+        Some(response)
+    } else {
+        None
+    }
```

Comment:
```rust
decimal::get_invariant(data_key).or(plurals::get_invariant(data_key))
```

Reply:
Probably `or_else`.

In the future, this will be chained with a lot of cases.  Is this clean and readable code?

```rust
decimal::get_invariant(data_key)
    .or_else(|| plurals.get_invariant(data_key))
    .or_else(|| datetime.get_invariant(data_key))
    .or_else(|| timezone.get_invariant(data_key))
    .or_else(|| regions.get_invariant(data_key))
    .or_else(|| unicode.get_invariant(data_key))
    .or_else(|| testing.get_invariant(data_key))
    // ...
```

Reply:
yeah, that looks really readable!